drivers/builtin/src/pk.c

```diff
@@ -1318,22 +1319,47 @@ int mbedtls_pk_sign_ext(mbedtls_pk_sigalg_t pk_type,
 int mbedtls_pk_check_pair(const mbedtls_pk_context *pub,
                           const mbedtls_pk_context *prv)
 {
+    psa_status_t status;
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    uint8_t *prv_key_buf = NULL;
+    size_t prv_key_len;
+    size_t prv_key_size;
+    mbedtls_svc_key_id_t key_id = prv->priv_id;
+
     if (pub->pk_info == NULL ||
         prv->pk_info == NULL) {
         return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
     }
 
-    if (prv->pk_info->check_pair_func == NULL) {
-        return MBEDTLS_ERR_PK_FEATURE_UNAVAILABLE;
-    }
-
     if ((prv->pk_info->type != MBEDTLS_PK_OPAQUE) &&
-        (pub->pk_info != prv->pk_info)) {
+        (pub->pk_info != prv->pk_info) &&
+        (pub->pk_info->type != prv->pk_info->type)) {
         return MBEDTLS_ERR_PK_TYPE_MISMATCH;
     }
 
-    return prv->pk_info->check_pair_func((mbedtls_pk_context *) pub,
-                                         (mbedtls_pk_context *) prv);
+    if ((prv->pk_info->type == MBEDTLS_PK_RSA) || (prv->pk_info->type == MBEDTLS_PK_RSASSA_PSS)) {
```

Comment:
Since Mbed-TLS/TF-PSA-Crypto#532 has now been merged, could you change this to use `mbedtls_pk_get_key_type()`?
```suggestion
    if (PSA_KEY_TYPE_IS_RSA(mbedtls_pk_get_key_type(prv))) {